scalafix-cli/src/main/scala/scalafix/cli/Cli.scala

```diff
@@ -101,9 +101,19 @@ object Cli {
   def zshNames: String =
     ScalafixRules.allNames.map(x => s""""$x"""").mkString(" \\\n  ")
 
+  def zshNamesDescriptions: String =
```

Comment:
You can try this out with

- `cli/assembly`
- create `scalafix` script that invokes `jar -jar cli/target/scala-2.12/scalafix.jar $@` (can't remember exact path)
- install the bash/zsh completions according to --help
- manually test tab completion with `scalafix --h<TAB>`

Not perfect, but that's the workflow I use :)

Reply:
PS I noticed that the --help message makes the --zsh/--bash instructions format weirdly

Reply:
Great! Should I try to improve the formatting and add them then? 
For zsh the output in the terminal is:
```
$ ./my-scalafix --zsh

#compdef scalafix
typeset -A opt_args
local context state line

_rule_names () {
   compadd "DottyKeywords - Rewrite that replaces enum and inline with `enum` and `inline` for compatibility with Dotty" \
  "DisableSyntax - " \
  "NoValInForComprehension - Rewrite that removes redundant val inside for-comprehensions" \
  "RemoveXmlLiterals - Rewrite that converts xml literals into interpolators for use with densh/scala-xml-quote" \
  "VolatileLazyVal - Rewrite all lazy vals to Dotty's volatile ones for safe publishing (default semantics of pre-Dotty Scala)" \
  "ProcedureSyntax - Rewrite that inserts explicit : Unit = for soon-to-be-deprecated procedure syntax def foo { ... }" \
  "ExplicitUnit - Rewrite that inserts inferred : Unit type annotation for abstract def foo" \
  "DottyVolatileLazyVal - Rewrite all lazy vals to Dotty's volatile ones for safe publishing (default semantics of pre-Dotty Scala)" \
  "DottyVarArgPattern - Rewrite to convert :_* vararg pattern syntax to @ syntax supported in Dotty." \
  "NoInfer - Linter for types that the Scala compiler cannot infer." \
  "Sbt1 - Rewrite to convert sbt 0.12 deprecated operators like <++= into sbt 0.13 :=/.value syntax" \
  "ExplicitResultTypes - Rewrite that inserts explicit type annotations for def/val/var" \
  "ExplicitReturnTypes - Rewrite that inserts explicit type annotations for def/val/var" \
  "RemoveUnusedImports - Rewrite that removes unused imports reported by the compiler under -Xwarn-unused-import." \
  "RemoveUnusedTerms - Rewrite that removes unused locals/privates by -Ywarn-unused:locals/privates" \
  "NoAutoTupling - Rewrite that inserts explicit tuples for adapted argument lists for compatibility with -Yno-adapted-args" \
  "Disable - Linter that reports an error on a configurable set of symbols."
}

local -a scalafix_opts
scalafix_opts=(
  "--usage[Print usage and exit]" \
   "--help[Print help message and exit]" \
   "-h[Print help message and exit]" \
   "--version[Print version number and exit]" \
   "-v[Print version number and exit]" \
   "--verbose[If set, print out debugging inforation to stderr.]" \
   "--config[File path to a .scalafix.conf configuration file.]" \
   "-c[File path to a .scalafix.conf configuration file.]" \
   "--config-str[String representing scalafix configuration]" \
   "-c[String representing scalafix configuration]" \
   "--sourceroot[Absolute path passed to semanticdb with 	-P\:semanticdb\:sourceroot\:<path>. Relative filenames persisted in the 	Semantic DB are absolutized by the sourceroot. Defaults to current 	working directory if not provided.]" \
   "--classpath[java.io.File.pathSeparator separated list of directories or jars 	containing '.semanticdb' files. The 'semanticdb' files are emitted by 	the semanticdb-scalac compiler plugin and are necessary for semantic 	rules like ExplicitResultTypes to function.]" \
   "--classpath-auto-roots[Automatically infer --classpath starting from these directories. 	Ignored if --classpath is provided.]" \
   "--tool-classpath[Additional classpath to use when classloading/compiling rules]" \
   "--no-strict-semanticdb[Disable validation when loading semanticdb files.]" \
   "*--rules=[Scalafix rules to run.]:rule:_rule_names" \
   "*-r=[Scalafix rules to run.]:rule:_rule_names" \
   "--stdout[If set, print fix to stdout instead of writing to file.]" \
   "--test[Exit non-zero code if files have not been fixed. Won't write to files.]" \
   "--out-from[Regex that is passed as first argument to 	fileToFix.replaceAll(outFrom, outTo)]" \
   "--out-to[Replacement string that is passed as second argument to 	fileToFix.replaceAll(outFrom, outTo)]" \
   "--exclude[Space separated list of regexes to exclude which files to fix. If a 	file match one of the exclude regexes, then it will not get fixed. 	Defaults to excluding no files.]" \
   "--single-thread[If true, run on single thread. If false (default), use all available 	cores]" \
   "--no-sys-exit[If true, does not sys.exit at the end. Useful for example in 	sbt-scalafix]" \
   "--quiet-parse-errors[Don't report parse errors for non-explictly passed filepaths.]" \
   "--bash[Print out bash completion file for scalafix. To install on scalafix 	--bash > /usr/local/etc/bash_completion.d/scalafix # Mac scalafix 	--bash > /etc/bash_completion.d/scalafix # Linux]" \
   "--zsh[Print out zsh completion file for scalafix. To install\: scalafix --zsh 	> /usr/local/share/zsh/site-functions/_scalafix]" \
   "--non-interactive[Don't use fancy progress bar.]" \
   "--project-id[String ID to prefix reported messages with]"
)

case $words[$CURRENT] in
      *) _arguments $scalafix_opts "*::filename:_files";;
esac

return 0
```
By the way, I am not sure I understand your PS: do you think my changes introduced a bug in how they appear?

Reply:
Hmmm. I propose we remove sbtNamesDescriptions and zshNamesDescriptions